clang/include/clang/Driver/Options.td

```diff
@@ -5107,7 +5107,13 @@ def : Flag<["-"], "fno-sycl-explicit-simd">,
   Flags<[CoreOption, Deprecated]>,
   Group<clang_ignored_legacy_options_Group>,
   HelpText<"Disable SYCL explicit SIMD extension. (deprecated)">;
-defm sycl_early_optimizations : OptOutCC1FFlag<"sycl-early-optimizations", "Enable", "Disable", " standard optimization pipeline for SYCL device compiler", [CoreOption]>;
+def fsycl_early_optimizations : Flag<["-"], "fsycl-early-optimizations">,
+  Flags<[CoreOption]>, Group<f_Group>,
+  HelpText<"Enable standard optimization pipeline for SYCL device compiler">;
+def fno_sycl_early_optimizations : Flag<["-"], "fno-sycl-early-optimizations">,
```

Comment:
I initially wanted the marshalling flag to just be on the negative variant, but I've just tested and if I add this flag onto the end of the previous definition OptOutCC1FFlag used, it also works. If adding it onto the old definition is better, I'll commit that change instead of this one.

Reply:
I'm not very familiar with that component of clang, but it seems to me that there is a neat and short way to describe both (on/off) flags using a single statement. I don't fully understand what is the reason not to use this functionality, because it allows to have less code duplication.

I will leave the final decision for clang driver reviewers, but I would prefer to modify the existing single statement instead of splitting it into two, if that's possible.

Reply:
If adding the marshalling info to the existing variant worked as expected, I would prefer to stick with that as opposed to splitting.

Reply:
Do you need the positive variant of this flag? If so,  does this work ?

```
defm sycl_early_optimizations : BoolFOption<"sycl_early_optimizations",
  CodeGenOpts<"DisableSYCLEarlyOpts">, DefaultFalse,
  PosFlag<SetFalse, [], "Enable">, NegFlag<SetTrue, [], "Disable">,
  BothFlags<[CC1Option, CoreOption], "standard optimization pipeline for SYCL device compiler">>;
```

I don't see why the positive variant is required though.

Reply:
I've pushed the revision that uses the old definition with the marshalling flag added on.

As this PR is for reimplementing the existing flag, I'm going to resolve this conversation for now, and I think deprecating/removing the positive variant should be left for now and in a separate future PR.